R/drive_mime_type.R

```diff
@@ -13,21 +12,33 @@
 #' ## get the mime type for jpegs
 #' drive_mime_type("jpeg")
 #' @export
-drive_mime_type <- function(type = NULL, verbose = TRUE) {
+drive_mime_type <- function(type = NULL) {
 
-  if (!(is.character(type) && length(type) == 1)) {
+  if (!(is.character(type))) {
     stop("Please update `type` to be a character string.", call. = FALSE)
   }
+  if (is.null(type)) {
+    return(.drive$mime_tbl)
+  }
+  m <- match(
+    type,
+    .drive$mime_tbl$human_type,
+    nomatch = NA_character_,
+    incomparables = NA
+  )
+  m <- ifelse(is.na(m),
+              match(type,
+                    .drive$mime_tbl$mime_type,
+                    nomatch = NA_character_,
+                    incomparables = NA
+              ),
+              m
+  )
+  mime_type <- .drive$mime_tbl$mime_type[m]
 
-  mime_type <- .drive$mime_tbl$mime_type[.drive$mime_tbl$mime_type == type |
-                                           (!is.na(.drive$mime_tbl$human_type) &
-                                              (.drive$mime_tbl$human_type == type))
-                                         ]
-  if (length(mime_type) == 0L) {
-    if (verbose) {
-      message(glue::glue("Ignoring `type` input. We do not have a mime type for files of type: {sq(type)}"))
-    }
-    return(invisible(NULL))
+  if (all(is.na(mime_type))) {
```

Comment:
Nice: this ends up being a good solution where we get to vectorize this, but can still use it easily inside functions that work on a dribble holding one file.